cmake/compiler/gcc/target.cmake

```diff
@@ -1,8 +1,11 @@
-set_ifndef(CC gcc)
 set_ifndef(C++ g++)
 
 # Configures CMake for using GCC, this script is re-used by several
 # GCC-based toolchains
+
+unset(CMAKE_C_COMPILER CACHE)
+unset(CMAKE_C_COMPILER)
```

Comment:
These unsets would be needed in every `target.cmake`. Factor out so it becomes shared.